src/transformers/models/visual_bert/modeling_visual_bert.py

```diff
@@ -0,0 +1,2122 @@
+            visual_token_type_embeddings = self.visual_token_type_embeddings(visual_token_type_ids)
+
+            if image_text_alignment is not None:
+
+                # TO-DO: Find a way to handle this in a better way.
+                # image_text_alignment = Batch x image_length x alignment_number. Each element denotes the position of the word corresponding to the image feature. -1 is the padding value.
+                image_text_alignment_mask = (image_text_alignment != -1).long()
+                # Get rid of the -1.
+                image_text_alignment = image_text_alignment_mask * image_text_alignment
+
+                # Batch x image_length x alignment length x dim
+                visual_position_embeddings = self.position_embeddings(
+                    image_text_alignment
+                ) * image_text_alignment_mask.to(dtype=next(self.parameters()).dtype).unsqueeze(-1)
+                visual_position_embeddings = visual_position_embeddings.sum(2)
+
+                # We want to averge along the alignment_number dimension.
+                image_text_alignment_mask = image_text_alignment_mask.to(dtype=next(self.parameters()).dtype).sum(2)
+                image_text_alignment_mask[image_text_alignment_mask == 0] = 1  # Avoid devide by zero error
+                visual_position_embeddings = visual_position_embeddings / image_text_alignment_mask.unsqueeze(-1)
+
+                visual_position_ids = torch.zeros(
+                    *visual_embeds.size()[:-1], dtype=torch.long, device=visual_embeds.device
+                )  # They use .cuda() but I believe this will be same as visual_embeds device.
+
+                # When fine-tuning the detector , the image_text_alignment is sometimes padded too long.
+                if visual_position_embeddings.size(1) != visual_embeds.size(1):
+                    assert visual_position_embeddings.size(1) >= visual_embeds.size(1)
+                    visual_position_embeddings = visual_position_embeddings[:, : visual_embeds.size(1), :]
+
+                visual_position_embeddings = visual_position_embeddings + self.visual_position_embeddings(
+                    visual_position_ids
+                )
+            else:
+                visual_position_ids = torch.zeros(
+                    *visual_embeds.size()[:-1], dtype=torch.long, device=visual_embeds.device
+                )  # They use .cuda() but I believe this will be same as visual_embeds device.
+                visual_position_embeddings = self.visual_position_embeddings(visual_position_ids)
+
+            visual_embeddings = visual_embeds + visual_position_embeddings + visual_token_type_embeddings
+
+            # Concate the two:
+            embeddings = torch.cat((embeddings, visual_embeddings), dim=1)  # concat the visual embeddings
+
+        embeddings = self.LayerNorm(embeddings)
+        embeddings = self.dropout(embeddings)
+        return embeddings
+
+
+class VisualBertSelfAttention(nn.Module):
```

Comment:
@patil-suraj This is actually `BertSelfAttention` rebranded as `VisualBertSelfAttention`, do you mean for the embeddings?

Reply:
If it's renamed it could benefit from having the following statement above:
```
# Copied from transformers.models.bert.modeling_bert.BertSelfAttention with Bert->VisualBertSelfAttention
```
this way we can make sure that the two implementations do not diverge

Reply:
If I add "copied from" to VisualBertSelfAttention, it also copies decoder-related arguments which we do not need, right?